File: mkbsd.js
const fs = require('fs');
const path = require('path');

// Function to delay the downloads
const delay = (ms) => new Promise(resolve => setTimeout(resolve, ms));

async function main() {
    const jsonFilePath = path.join(__dirname, 'images.json');
    
    // Load local JSON file
    let jsonData;
    try {
        if (!fs.existsSync(jsonFilePath)) {
            throw new Error('⛔ JSON file not found.');
        }

        const jsonFileContent = fs.readFileSync(jsonFilePath, 'utf-8');
        jsonData = JSON.parse(jsonFileContent);
        console.info('📂 JSON file loaded successfully!');
    } catch (error) {
        console.error(`⛔ Failed to load JSON file: ${error.message}`);
        return;
    }

    const data = jsonData.data;
    if (!data) {
        console.error('⛔ JSON does not have a "data" property at its root.');
        return;
    }

    // Loop through each item in the JSON
    for (const key in data) {
        const subproperty = data[key];
        if (subproperty && subproperty.dhd) {
            const imageUrl = subproperty.dhd;
            console.info(`🔍 Found image URL!`);

            // Extract the artist name before the underscore
            const artistNameMatch = imageUrl.match(/a~([^_/]+)/);
            const artistName = artistNameMatch ? artistNameMatch[1] : 'unknown_artist';
            const artistDir = path.join(__dirname, 'downloads', artistName);

            // Create artist directory if it doesn't exist
            if (!fs.existsSync(artistDir)) {
                fs.mkdirSync(artistDir, { recursive: true });
                console.info(`📁 Created directory: ${artistDir}`);
            }

            // Extract the filename and extension
            const urlPath = new URL(imageUrl).pathname;
            const fileName = path.basename(urlPath); // Filename including extension (e.g., .jpg or .png)
            const filePath = path.join(artistDir, fileName);

            // Download the image and save it to the specified path
            try {
                await downloadImage(imageUrl, filePath);
                console.info(`🖼️ Saved image to ${filePath}`);
                await delay(250); // Delay between downloads
            } catch (err) {
                console.error(`❌ Error downloading image: ${err.message}`);
            }
        }
    }
}

// Function to download the image from the provided URL
async function downloadImage(url, filePath) {
    const response = await fetch(url);
    if (!response.ok) {
        throw new Error(`Failed to download image: ${response.statusText}`);
    }
    const arrayBuffer = await response.arrayBuffer();
    const buffer = Buffer.from(arrayBuffer);
    await fs.promises.writeFile(filePath, buffer);
}

// ASCII art function (optional)
function asciiArt() {
    console.info(`
 /$$      /$$ /$$   /$$ /$$$$$$$   /$$$$$$  /$$$$$$$
| $$$    /$$$| $$  /$$/| $$__  $$ /$$__  $$| $$__  $$
| $$$$  /$$$$| $$ /$$/ | $$  \\ $$| $$  \\__/| $$  \\ $$
| $$ $$/$$ $$| $$$$$/  | $$$$$$$ |  $$$$$$ | $$  | $$
| $$  $$$| $$| $$  $$  | $$__  $$ \\____  $$| $$  | $$
| $$\\  $ | $$| $$\\  $$ | $$  \\ $$ /$$  \\ $$| $$  | $$
| $$ \\/  | $$| $$ \\  $$| $$$$$$$/|  $$$$$$/| $$$$$$$/
|__/     |__/|__/  \\__/|_______/  \\______/ |_______/`);
    console.info(`🤑 Starting downloads from your favorite sellout grifter's wallpaper app...`);
}

// Start the program with ASCII art and delay
(() => {
    asciiArt();
    setTimeout(main, 5000);
})();

File: mkbsd.py
import os
import time
import aiohttp
import asyncio
import json
from urllib.parse import urlparse

# Load the local JSON file
json_file_path = 'images.json'

async def delay(ms):
    await asyncio.sleep(ms / 1000)

async def download_image(session, image_url, file_path):
    try:
        async with session.get(image_url) as response:
            if response.status != 200:
                raise Exception(f"Failed to download image: {response.status}")
            content = await response.read()
            with open(file_path, 'wb') as f:
                f.write(content)
    except Exception as e:
        print(f"Error downloading image: {str(e)}")

async def main():
    try:
        # Load JSON data from the local file
        with open(json_file_path, 'r') as json_file:
            json_data = json.load(json_file)

        data = json_data.get('data')
        
        if not data:
            raise Exception('⛔ JSON does not have a "data" property at its root.')

        async with aiohttp.ClientSession() as session:
            for key, subproperty in data.items():
                if subproperty and subproperty.get('dhd'):
                    image_url = subproperty['dhd']
                    print(f"🔍 Found image URL!")
                    
                    # Extract artist name from the URL
                    artist_name = image_url.split('a~')[1].split('_')[0]
                    artist_dir = os.path.join(os.getcwd(), 'downloads', artist_name)

                    if not os.path.exists(artist_dir):
                        os.makedirs(artist_dir)
                        print(f"📁 Created directory: {artist_dir}")

                    # Extract filename from the URL
                    filename = os.path.basename(urlparse(image_url).path)  # Name including extension
                    file_path = os.path.join(artist_dir, filename)

                    await download_image(session, image_url, file_path)
                    print(f"🖼️ Saved image to {file_path}")
                    
                    await delay(250)

    except Exception as e:
        print(f"Error: {str(e)}")

def ascii_art():
    print("""
 /$$      /$$ /$$   /$$ /$$$$$$$   /$$$$$$  /$$$$$$$
| $$$    /$$$| $$  /$$/| $$__  $$ /$$__  $$| $$__  $$
| $$$$  /$$$$| $$ /$$/ | $$  \\ $$| $$  \\__/| $$  \\ $$
| $$ $$/$$ $$| $$$$$/  | $$$$$$$ |  $$$$$$ | $$  | $$
| $$  $$$| $$| $$  $$  | $$__  $$ \\____  $$| $$  | $$
| $$\\  $ | $$| $$\\  $$ | $$  \\ $$ /$$  \\ $$| $$  | $$
| $$ \\/  | $$| $$ \\  $$| $$$$$$$/|  $$$$$$/| $$$$$$$/
|__/     |__/|__/  \\__/|_______/  \\______/ |_______/""")
    print("")
    print("🤑 Starting downloads from your favorite sellout grifter's wallpaper app...")

if __name__ == "__main__":
    ascii_art()
    time.sleep(5)
    asyncio.run(main())